hibernate-core/src/main/java/org/hibernate/engine/jdbc/internal/ResultSetReturnImpl.java

```diff
@@ -61,6 +65,7 @@ public ResultSet extract(PreparedStatement statement) {
 			}
 			finally {
 				jdbcExecuteStatementEnd();
+				sqlStatementLogger.logSlowQuery( statement.toString(), executeStart );
```

Comment:
could you make sure that this is not executed if the feature is not enabled?

In particular doing statement.toString() is otherwise going to be a performance problem.

(same for the other methods)

Reply:
The method will be executed, but immediately return if the feature is not enabled. 

statement.toString() execution is fixed in 35a2d98abd80e3ed64734e22f69ec6277b08726a